### PR TITLE
Cleanup camas practice

### PR DESCRIPTION
**Pull request by natesales**

### CODE REVIEW

**Review by sanashah007 — APPROVED** [review]

File: src/main/java/org/team1540/robot2022/Robot.java [hunk 1]
```diff
@@ -67,16 +59,18 @@ public void robotPeriodic() {
         // block in order for anything in the Command-based framework to work.
         CommandScheduler.getInstance().run();
 
-        // Update the limelight's custom SmartDashboard values
-        limelight.updateSmartDashboardValues();
+        // Update the robotContainer.limelight's custom SmartDashboard values
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/Robot.java:62]:
```suggestion
        // Update the limelight's custom SmartDashboard values
```
Seems like find and replace included this one too

File: src/main/java/org/team1540/robot2022/commands/climber/Climber.java [hunk 2]
```diff
@@ -2,42 +2,52 @@
...
-    private TalonFX motorRight = new TalonFX(ClimberConstants.Motors.right);
-    
-    private final Solenoid solenoidLeft = new Solenoid(Constants.ph, PneumaticsModuleType.REVPH, ClimberConstants.Solenoids.left);
-    private final Solenoid solenoidRight = new Solenoid(Constants.ph, PneumaticsModuleType.REVPH, ClimberConstants.Solenoids.right);
+    private final ChickenTalonFX motorLeft = new ChickenTalonFX(ClimberConstants.Motors.LEFT);
+    private final ChickenTalonFX motorRight = new ChickenTalonFX(ClimberConstants.Motors.RIGHT);
+    private final ChickenTalonFX[] motors = new ChickenTalonFX[]{motorLeft, motorRight};
+
+    private final Solenoid solenoidA = new Solenoid(Constants.PNEUMATIC_HUB, PneumaticsModuleType.REVPH, ClimberConstants.Solenoids.SOLENOID_A);
+    private final Solenoid solenoidB = new Solenoid(Constants.PNEUMATIC_HUB, PneumaticsModuleType.REVPH, ClimberConstants.Solenoids.SOLENOID_B);
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/climber/Climber.java:20]: These should probably be combined as a `DoubleSolenoid`

File: src/main/java/org/team1540/robot2022/commands/climber/Climber.java [hunk 3]
```diff
@@ -51,4 +61,11 @@ private void updatePIDs() {
         motorLeft.config_kP(0, SmartDashboard.getNumber("climber/PID/kP", 0.3));
     }
 
+    /**
+     * Sets the NeutralMode for the drivetrain (either coast or brake)
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/climber/Climber.java:65]:
```suggestion
     * Sets the NeutralMode for the climber (either coast or brake)
```

File: src/main/java/org/team1540/robot2022/commands/drivetrain/Drivetrain.java [hunk 4]
```diff
@@ -154,6 +175,24 @@ public void setVolts(double leftVolts, double rightVolts) {
         driveRFront.setVoltage(rightVolts);
     }
 
+    /**
+     * Set motor voltages from velocity setpoints
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/drivetrain/Drivetrain.java:179]:
```suggestion
     * Sets motor voltages from velocity setpoints
```

File: src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java [hunk 5]
```diff
@@ -21,7 +21,11 @@ public OdometryResetSequence(Drivetrain drivetrain, NavX navx, Limelight limelig
                 new WaitCommand(0.5),
                 new InstantCommand(() -> limelight.setLeds(initialLightState))
             ),
-            new InstantCommand(() -> drivetrain.resetOdometry(zeroPose)) // Reset odometry to known starting pose
+            sequence(
+                new InstantCommand(navx::zeroYaw), // Reset odometry to known starting pose
+                new InstantCommand(() -> drivetrain.resetOdometry(zeroPose)) // Reset odometry to known starting pose
+            ),
+            new InstantCommand(() -> drivetrain.setNeutralMode(NeutralMode.Coast)) // Reset odometry to known starting pose
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java:28]:
```suggestion
                new InstantCommand(navx::zeroYaw),
                new InstantCommand(() -> drivetrain.resetOdometry(zeroPose)) // Reset odometry to known starting pose
            ),
            new InstantCommand(() -> drivetrain.setNeutralMode(NeutralMode.Coast)) // Set drivetrain to coast for easier dragging
```

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java [hunk 6]
```diff
@@ -191,25 +194,14 @@ public Command commandStop() {
...
-    }
-    /**
-     * Returns a command to start the indexing listeners and motors
+     * Sets the NeutralMode for the drivetrain (either coast or brake)
      *
-     * @return an InstantCommand to set the standby mode
+     * @param mode The mode to set the wheels to
      */
-    public Command commandStart() {
-        return new InstantCommand(this::start);
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:212]: Why was this removed?

File: src/main/java/org/team1540/robot2022/commands/intake/Intake.java [hunk 7]
```diff
@@ -51,10 +57,20 @@ public Command commandStop() {
...
+     *
      * @param isUp if the indexer is folded up
      * @return An InstantCommand
      */
     public Command commandSetFold(boolean isUp) {
         return new InstantCommand(() -> this.setFold(isUp));
     }
+
+    /**
+     * Sets the NeutralMode for the drivetrain (either coast or brake)
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/intake/Intake.java:69]:
```suggestion
     * Sets the NeutralMode for the indexer (either coast or brake)
```

File: src/main/java/org/team1540/robot2022/commands/intake/IntakeSequence.java [hunk 8]
```diff
@@ -1,25 +1,36 @@
...
+        addRequirements(intake, indexer, shooter);
         addCommands(
-                indexer.commandStart(),
-                new IntakeSpinCommand(intake, indexer, Constants.IntakeConstants.speed)
-                
+                new InstantCommand(() -> {
+                    indexer.setStandby(false);
+                    if (indexer.isFull()) {
+                        indexer.set(Indexer.IndexerState.OFF, Indexer.IndexerState.OFF);
+                        shooter.commandSetVelocity(InterpolationTable.shooterSpinupFront, InterpolationTable.shooterSpinupRear);
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/intake/IntakeSequence.java:23]: Why is the intake sequence turning on the shooter?

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java [hunk 9]
```diff
@@ -2,25 +2,24 @@
...
+import org.team1540.robot2022.Constants.IndexerConstants.IndexerMotors;
+import org.team1540.robot2022.utils.ChickenTalonFX;
 
 public class Indexer extends SubsystemBase {
-    private final TalonFX bottomMotor = new TalonFX(IndexerMotors.bottomMotor);
-    private final TalonFX topMotor = new TalonFX(IndexerMotors.topMotor);
-    private final TalonFX[] motors = {topMotor, bottomMotor};
+    private final ChickenTalonFX bottomMotor = new ChickenTalonFX(IndexerMotors.BOTTOM_MOTOR);
+    private final ChickenTalonFX topMotor = new ChickenTalonFX(IndexerMotors.TOP_MOTOR);
+    private final ChickenTalonFX[] motors = {topMotor, bottomMotor};
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:19]: Why are these changing to `ChickenTalonFX`s? I don't see anything using those methods.

File: src/main/java/org/team1540/robot2022/utils/Limelight.java [hunk 10]
```diff
@@ -91,7 +91,7 @@ public boolean getLeds() {
      */
     public void setLeds(boolean isOn) {
         if (getLeds() != isOn) {
-            limelightTable.getEntry("ledMode").setNumber(isOn ? 0 : 1);
+            limelightTable.getEntry("ledMode").setNumber(0);
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/utils/Limelight.java:94]: This should probably be reverted

File: src/main/java/org/team1540/robot2022/commands/intake/IntakeSequence.java (hunk 8 above)
Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/intake/IntakeSequence.java:23]: This is to spin up the shooter when the indexer is full, instead of relying on a copilot button.

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java (hunk 9 above)
Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:19]: I changed these for consistency. I think we should make every TalonFX into a ChickenTalonFX (especially for voltage based kinematics)

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java (hunk 6 above)
Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:212]: I moved this to an InstantCommand to avoid having to pass the shooter subsystem in here.

File: src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java (hunk 5 above)
Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java:28]: This `NeutralMode` control is now redundant since there are `Trigger`s that set mechanism brakes.

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java (hunk 9 above)
Comment (rutmanz) [reply src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:19]: Then the rest of the subsystems should probably be updated too

File: src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java (hunk 5 above)
Comment (rutmanz) [reply src/main/java/org/team1540/robot2022/commands/drivetrain/OdometryResetSequence.java:28]:
```suggestion
                new InstantCommand(navx::zeroYaw),
                new InstantCommand(() -> drivetrain.resetOdometry(zeroPose)) // Reset odometry to known starting pose
            )
```

File: src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java (hunk 9 above)
Comment (rutmanz) [reply src/main/java/org/team1540/robot2022/commands/indexer/Indexer.java:19]: Specifically Intake and Shooter

**Review by rutmanz — APPROVED** [review]
